### DISCUSSION TITLE
.flat() issue on iOS 11

### DISCUSSION THREAD

**Discussion by tophergd** [react-hook-form/react-hook-form#3991] [General] [answered]
Hello folks, I'm looking for a bit of guidance on this compatibility issue. 

I have an error surfacing on Safari v11 (iOS) that is showing `undefined is not a function` on a usage of `.flat()` from within react-hook-form. This makes sense since `.flat()` is not supported until iOS safari v12. 

I assumed that since react-hook-form <7 has support for IE11 then the usage of `.flat()` should be covered within that.

Maybe I'm missing something but any help would be much appreciated :)!

Thanks in advance!

**Comment by bluebill1049** [comment, marked as answer]
you want to include corejs for this fix?

**Reply by tophergd** [reply]
Thanks i've added a .flat() polyfill from core-js to sort this. 

Also raised https://github.com/react-hook-form/documentation/pull/508 to add this as a required polyfill in the docs.

thanks for the help 👍

**Reply by bluebill1049** [reply]
thanks a lot for that 🙏
